projects/ADuCM3029_demo_cn0397/src/cn0397_app.cpp

```diff
@@ -0,0 +1,519 @@
+ *             should just be used for flags and semaphores.
+ */
+static void ApplicationCallback(void * pCBParam, uint32_t Event, void * pArg)
+{
+    switch(Event)
+    {
+        case GAP_EVENT_CONNECTED:
+            DEBUG_MESSAGE("Connected!\r\n");
+            gbConnected = true;
+            break;
+
+        case GAP_EVENT_DISCONNECTED:
+            DEBUG_MESSAGE("Disconnected!\r\n");
+            geMode      = ADI_BLE_GAP_MODE_NOTCONNECTABLE;
+            gbConnected = false;
+            reg_flag = 0;
+            break;
+
+        case DATA_EXCHANGE_RX_EVENT:
+            DEBUG_MESSAGE("Data received!\r\n");
+            break;
+
+        case DATA_EXCHANGE_TX_COMPLETE:
+        	DEBUG_MESSAGE(("Data sent!\r\n"));
+        	if (reg_flag==0)
+        		reg_flag=1;
+            break;
+
+        case GAP_EVENT_MODE_CHANGE:
+            DEBUG_MESSAGE("GAP mode changed.\r\n");
+            break;
+
+        case GAP_EVENT_CONNECTION_UPDATED:
+            DEBUG_MESSAGE("Connection interval updated.\r\n");
+            break;
+
+        case BLE_RADIO_ERROR_READING:
+            /* If you want to enable this print statement, please be aware that the first
+             * packet sent from the radio on startup will cause this error. It is a known bug
+             * and will not have any adverse effects on the application.
+             *
+             *DEBUG_MESSAGE("Failed to read a packet from the radio.\r\n");
+             *
+             */
+        	reg_flag = 0;
+            break;
+
+        case BLE_RADIO_ERROR_PARSING:
+        	DEBUG_MESSAGE(("Failed to parse a packet from the radio.\r\n"));
+        	reg_flag = 0;
+            break;
+
+        case BLE_RADIO_ERROR_PROCESSING:
+        	DEBUG_MESSAGE(("Failed to process a packet from the radio.\r\n"));
+        	reg_flag = 0;
+            break;
+
+        default:
+        	DEBUG_MESSAGE(("Unexpected event received.\r\n"));
+            break;
+    }
+}
+
+/*!
+ * @brief      Set Advertising Mode
+ *
+ * @details    Helper function to avoid repeated code in main.
+ *
+ */
+static void SetAdvertisingMode(void)
+{
+    ADI_BLER_RESULT eResult;
+
+    eResult = adi_radio_SetMode(PERIPHERAL_ADV_MODE, 0u, 0u);
+    DEBUG_RESULT("Error setting the mode.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    eResult = adi_ble_WaitForEventWithTimeout(GAP_EVENT_MODE_CHANGE, 5000u);
+    DEBUG_RESULT("Error waiting for GAP_EVENT_MODE_CHANGE.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    eResult = adi_radio_GetMode(&geMode);
+    DEBUG_RESULT("Error getting the mode.\r\n", eResult, ADI_BLER_SUCCESS);
+
+    if(geMode != PERIPHERAL_ADV_MODE) {
+        DEBUG_MESSAGE("Error in SetAdvertisingMode.\r\n");
+    }
+}
+
+/*!
+ * @brief      Initializes the bluetooth
+ *
+ * @details    Data Exchange profile is initialized to send
+ *             data to the connected central device.
+ */
+static void InitBluetoothLowEnergy(void)
+{
+    ADI_BLER_RESULT eResult;
```

Comment:
In Release mode this variable is set but not used. We could use it to implement error checking if we want that.